### PR TITLE
Daemon API docs

### PR DESCRIPTION
**Pull request by VoidingWarranties**
Incomplete, see TODOs.

### CODE REVIEW

File: doc/API.md [hunk 1]
```diff
@@ -82,104 +82,61 @@ Table of contents
...
-Queries:
-
-* /daemon/constants [GET]
-* /daemon/stop      [GET]
-* /daemon/version   [GET]
+| Route                                     | HTTP verb |
+| ----------------------------------------- | --------- |
+| [/daemon/constants](#daemonconstants-get) | GET       |
+| [/daemon/stop](#daemonstop-get)           | GET       |
+| [/daemon/version](#daemonversion-get)     | GET       |
```
Comment (VoidingWarranties) [inline doc/API.md]: Should have a note linking to `Wallet.md`. See Gateway for an example.

Comment (DavidVorick) [inline doc/API.md]: to wallet.md? Isn't this the daemon PR?

File: doc/API.md [hunk 2]
```diff
@@ -82,104 +82,61 @@ Table of contents
...
-
-	siacoinprecision types.Currency (string)
+###### JSON Response [(with comments)](/doc/api/Daemon.md#json-response)
+```javascript
+{
+  "genesistimestamp":       1257894000, // Unix time
+  "blocksizelimit":         2000000,    // bytes
+  "blockfrequency":         600,        // seconds per block
+  "targetwindow":           1000,       // blocks
+  "mediantimestampwindow":  11,         // TODO: What are the units? blocks?
```
Comment (DavidVorick) [inline doc/API.md]: blocks

File: doc/API.md [hunk 3]
```diff
@@ -82,104 +82,61 @@ Table of contents
...
+  "futurethreshold":        10800,      // seconds
+  "extremefuturethreshold": 21600,      // seconds
+  "siafundcount":           "10000",
+  "siafundportion":         "39/1000",
+  "maturitydelay":          144,        // blocks
+
+  "initialcoinbase": 300000, // TODO: What are the units? SiaCoins? But not hastings? This is probably the only place we use a uint64 for currency and don't use hastings.
```
Comment (DavidVorick) [inline doc/API.md]: Siacoins. Definitely breaks from the convention of returning everything in hastings, but we have a compatibility promise that we need to uphold. So, I'd add some extra documentation explaining that, while this is presenting in siacoins and not hastings, it is unusual and doesn't follow the rest of the API.

File: doc/API.md [hunk 4]
```diff
@@ -82,104 +82,61 @@ Table of contents
...
+  "minimumcoinbase": 30000,  // TODO: See above TODO.
```
Comment (DavidVorick) [inline doc/API.md]: same deal

File: doc/api/Daemon.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  // Fraction of each file contract payout given to siafund holders.
+  "siafundportion": "39/1000",
+  // Number of children a block must have before it is considered "mature."
+  "maturitydelay": 144, // blocks
+
+  // Number of coins given to the miner of the first block.
+  "initialcoinbase": 300000, // TODO: What are the units? SiaCoins? But not hastings? This is probably the only place we use a uint64 for currency and don't use hastings.
+  // Minimum number of coins paid out to the miner of a block (the coinbase
+  // decreases with each block).
+  "minimumcoinbase": 30000, // TODO: See above TODO.
```
Comment (DavidVorick) [inline doc/api/Daemon.md]: :+1:

File: doc/api/Daemon.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+  "extremefuturethreshold": 21600, // seconds
+  // Total number of siafunds.
+  "siafundcount": "10000",
+  // Fraction of each file contract payout given to siafund holders.
+  "siafundportion": "39/1000",
+  // Number of children a block must have before it is considered "mature."
+  "maturitydelay": 144, // blocks
+
+  // Number of coins given to the miner of the first block.
+  "initialcoinbase": 300000, // TODO: What are the units? SiaCoins? But not hastings? This is probably the only place we use a uint64 for currency and don't use hastings.
```
Comment (DavidVorick) [inline doc/api/Daemon.md]: :+1:

File: doc/API.md (hunk 1 above)
Comment (VoidingWarranties) [inline doc/API.md]: Ooops, right. I think I copy-pasted this from the other PR.

**Comment by VoidingWarranties** [comment]
@DavidVorick this PR adds the "extremefuturetimestamp" field. In #1389 you mentioned adding a field for v1.1.0. Should we wait to add fields to the API until minor releases?

**Comment by VoidingWarranties** [comment]
Rebased and updated. Good to go pending my above comment.

File: doc/api/Daemon.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+  "blocksizelimit": 2000000, // bytes
+  // Target for how frequently new blocks should be mined.
+  "blockfrequency": 600, // seconds per block
+  // Height of the window used to adjust the difficulty.
+  "targetwindow": 1000, // blocks
+  // Duration of the window used to adjust the difficulty.
+  "mediantimestampwindow": 11, // blocks
+  // How far in the future a block can be without being rejected.
+  "futurethreshold": 10800, // seconds
+  // TODO: how is this different from futurethreshold?
```
Comment (DavidVorick) [inline doc/api/Daemon.md]: TODO: document correctly :)

File: doc/api/Daemon.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+  "initialcoinbase": 300000, // SiaCoins.
+  // Minimum number of coins paid out to the miner of a block (the coinbase
+  // decreases with each block). Note that elsewhere in the API currency is
+  // typically returned in hastings and as a bignum. This is not the case
+  // here.
+  "minimumcoinbase": 30000, // SiaCoins
+
+  // Initial target.
+  "roottarget": [0,0,0,0,32,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],
+  // Initial depth.
```
Comment (DavidVorick) [inline doc/api/Daemon.md]: These feel incomplete, though I'm not sure how you'd expand the documentation here without getting too technical. Maybe 'Target hash for the first block following the genesis block' and 'Initial amount of work assigned to the blockchain.'

**Comment by DavidVorick** [comment]
Yeah, we should wait to extend the API until 1.1, let's pull out the code and create an issue that we insert into milestone v1.1

**Comment by VoidingWarranties** [comment]
Removed "extremefuturethreshold" field (to be added in another PR for v1.1.0), corrected "futurethreshold" comment, and rebased. Should be good to go.

**Pull request merged**